docs/content/docs/plugins/magic-link.mdx

```diff
@@ -134,7 +134,38 @@ and a `request` object as the second parameter.
 
 **expiresIn**: specifies the time in seconds after which the magic link will expire. The default value is `300` seconds (5 minutes).
 
-**disableSignUp**: If set to `true`, the user will not be able to sign up using the magic link. The default value is `false`.
+**disableSignUp**: Controls whether new users can sign up via magic link. Can be a boolean or a function that receives the request context and returns a boolean (sync or async). The default value is `false`.
+
+- When set to `true`, only existing users can sign in with magic links.
+- When set to a function, you can conditionally disable signups based on your application logic.
+- The function receives `Request | undefined` as a parameter (the request may be undefined in some contexts).
+
+Example with function handler:
+```ts
+magicLink({
+    sendMagicLink: async ({ email, token, url }, request) => {
+        // send email
+    },
+    disableSignUp: (request) => {
+        if (!request) return false;
+        const referer = request.headers.get("referer");
+        return referer?.includes("/admin") ?? false;
+    }
```

Comment:
From logical consideration, this is incorrect. You cannot trust data from http request, so disabling based on referer is insecure

Reply:
Agreed there, can definitely change the example. The use case we have right now is more of a UX consideration than blocking a potential attacker from spoofing the referer. ie. we want to check if the user is signing up from our sign-up page and not our sign-in page as the same `signIn.magicLink` is used for both when `disableSignUp` is `false`

Reply:
Updated the example to check blocked domain list instead along side the context feedback.
https://github.com/better-auth/better-auth/pull/5089/commits/5a48c8195a4c8dc81db9ae095a29eb828ec588e9